storm-core/src/jvm/org/apache/storm/cluster/PaceMakerStateStorage.java

```diff
@@ -159,10 +162,6 @@ public void set_worker_hb(String path, byte[] data, List<ACL> acls) {
                         ret = details;
                     }
                 }
-                if(ret == null) {
-                    throw new HBExecutionException("Failed to get a response.");
-                }
-                LOG.debug("Successful get_worker_hb");
                 return ret;
```

Comment:
I think that is best, if all of them return something invalid or we get nothing back at all then we throw.